Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+namespace BootCamp.Chapter
+{
+    public static class Lesson4
+    {
+        public static void Demo()
+        {
+            char repeatQuestionAndReportForAnotherUser = 'y';
+
+            while (repeatQuestionAndReportForAnotherUser == 'y')
+            {
+                string firstName = RequestString("Please enter first name:");
+
+                string surname = RequestString("Please enter last name:");
+
+                int age = RequestInt("Please enter age in years:");
+
+                float weightInKilograms = RequestFloat("Please enter weight in kilograms:");
+
+                float heightInMeters = RequestFloat("Please enter height in meters:");
+
+                ReportBasicInfo(firstName, surname, age, weightInKilograms, heightInMeters);
+
+                ReportBodyMassIndex(firstName, weightInKilograms, heightInMeters);
+
+                repeatQuestionAndReportForAnotherUser = RequestUserChoiceToRepeatOrNot();
+            }
+        }
+        public static string RequestString(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            return GetValidStringOrErrorCode();
+        }
+        private static string GetValidStringOrErrorCode()
+        {
+            var userInput = Console.ReadLine();
+            if (ReportTrueIfUserInputIsEmpty(userInput))
+            {
+                Console.Write($"Name cannot be empty.");
+                return "-";
+            }
+            return userInput;
+        }
+        public static int RequestInt(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            var userInput = Console.ReadLine();
+            if (ReportTrueIfUserInputIsEmpty(userInput))
+            {
+                return 0;
+            }
+            return GetValidIntOrErrorCode(userInput);
+        }
+        static int GetValidIntOrErrorCode(string input)
+        {
+            var isInt = int.TryParse(input, out var number);
+            if (!isInt)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+
+            return number;
+        }
+        public static float RequestFloat(string prompt)
+        {
+            Console.Write($"{prompt}{Environment.NewLine}");
+            var userInput = Console.ReadLine();
+            if (ReportTrueIfUserInputIsEmpty(userInput))
+            {
+                return 0;
+            }
+            return GetValidFloatOrErrorCode(userInput);
+        }
+        static float GetValidFloatOrErrorCode(string input)
+        {
+            var isFloat = float.TryParse(input, out var number);
+            if (!isFloat)
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
+            }
+            return number;
+        }
+        public static bool ReportTrueIfUserInputIsEmpty(string input)
```

Comment:
This function doesn't add anything to string.IsNullOrEmpty(input) and takes even more characters to write.
You should delete this one and use string.IsNullOrEmpty() instead.
If you think you have a good reason to do it that way feel free to explain your decision! :)

Reply:
I think I was trying to keep logic abstracted, but it just adds complexity. I'll take it out. Thanks.